opensrp-chw-core/src/main/java/org/smartregister/chw/core/rule/FpAlertRule.java

```diff
@@ -52,12 +52,19 @@ public boolean isCocPopValid(int dueDay, int overdueDate) {
 
     public boolean isCondomValid(int dueDay, int overdueDate) {
         if (lastVisitDate != null) {
-            if (Months.monthsBetween(new DateTime(lastVisitDate), new DateTime()).getMonths() <= 1) {
-                this.dueDate = new DateTime().withDayOfMonth(dueDay);
-                this.overDueDate = new DateTime().withDayOfMonth(overdueDate);
-            } else {
-                this.dueDate = lastVisitDate.withDayOfMonth(dueDay);
-                this.overDueDate = lastVisitDate.withDayOfMonth(overdueDate);
+            if (( new DateTime(lastVisitDate).getMonthOfYear() == DateTime.now().getMonthOfYear()) && ( new DateTime(lastVisitDate).getYear() == DateTime.now().getYear())){
+                this.dueDate = new DateTime().plusMonths(1).withDayOfMonth(dueDay);
+                this.overDueDate = new DateTime().plusMonths(1).withDayOfMonth(overdueDate);
+            }
+            else{
+                if(( new DateTime(lastVisitDate).getYear() == DateTime.now().getYear()) && ((DateTime.now().getMonthOfYear()) - (new DateTime(lastVisitDate).getMonthOfYear()) == 1)){
```

Comment:
Maybe the `new DateTime(lastVisitDate).getMonthOfYear()` and `new DateTime(lastVisitDate).getMonthOfYear()` can be referenced in variables to be used in the if statements to avoid repetition?